app/src/main/java/de/westnordost/streetcomplete/quests/construction/MarkCompletedHighwayConstruction.java

```diff
@@ -0,0 +1,106 @@
+package de.westnordost.streetcomplete.quests.construction;
+
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.text.TextUtils;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import de.westnordost.osmapi.map.data.BoundingBox;
+import de.westnordost.streetcomplete.R;
+import de.westnordost.streetcomplete.data.meta.OsmTaggings;
+import de.westnordost.streetcomplete.data.osm.changes.StringMapChangesBuilder;
+import de.westnordost.streetcomplete.data.osm.download.MapDataWithGeometryHandler;
+import de.westnordost.streetcomplete.data.osm.download.OverpassMapDataDao;
+import de.westnordost.streetcomplete.data.osm.tql.OverpassQLUtil;
+import de.westnordost.streetcomplete.quests.DateUtil;
+import de.westnordost.streetcomplete.quests.YesNoQuestAnswerFragment;
+
+public class MarkCompletedHighwayConstruction extends MarkCompletedConstruction
+{
+	@Inject
+	public MarkCompletedHighwayConstruction(OverpassMapDataDao overpassServer) {
+		super(overpassServer);
+	}
+
+	@Override public boolean download(BoundingBox bbox, MapDataWithGeometryHandler handler)
+	{
+		return overpassServer.getAndHandleQuota(getOverpassQuery(bbox), handler);
+	}
+
+	/** @return overpass query string to get streets marked as under construction but excluding ones
+	 * - with invalid construction tag
+	 * - with tagged opening date that is in future
+	 * - recently edited (includes adding/updating check_date tags)
+	 * - with fixme (as in some situations this quest may add fixme note as an edit)
+	 * . */
+	private String getOverpassQuery(BoundingBox bbox)
+	{
+		String groupName = ".roads_under_construction";
+		String resultsName = ".roads_for_review";
+		return OverpassQLUtil.getGlobalOverpassBBox(bbox) +
+			"way" + getQueryPart("highway", groupName, 14) +
+			"(" +
+			"  way[highway=construction][fixme];" +
+			") -> .with_fixme;" +
+			"(" +
+			groupName + " - .with_fixme;" +
+			") -> " + resultsName + ";" +
+			resultsName + " out meta geom;";
+	}
+
+	public void applyAnswerTo(Bundle answer, StringMapChangesBuilder changes)
+	{
+		if(answer.getBoolean(YesNoQuestAnswerFragment.ANSWER)) {
+			if(isAnyAccessTagPresent(changes)){
+				// there are some access tag present that are likely to be temporary
+				// and valid only during construction - marking construction as finished
+				// would likely cause nonobvious tagging mistake
+				changes.add("fixme", "construction is completed but there are access tags that maybe were true only during construction period");
```

Comment:
Hmm, so then this must be fixed/confirmed by someone else later who visits the place. It would be cool if there was an app that would automatically show these to-be-surveyed points to surveyors and let them solve them.

*Oh, wait!* ;-)

Couldn't this be covered within the app? If any access-like key is set, ask:
*"Is its use now restricted to certain road users?"*
[Restricted] [Not restricted]

On clicking "not restricted", the access tag(s) are deleted. If clicking "Yes", either:
1. the restriction tags that are currently set are simply not removed. IMO this would cover all cases or do you have examples that show otherwise?
2. a dialog could pop up that prompts the user to say how it is restricted. This info could be put into the fixme tag:
`"construction is complete but access tags may now be incorrect: [user message]"` (a tag value cannot have more than 255 characters, as far as I know.

Reply:
> if there was an app that would automatically show these to-be-surveyed points to surveyors

Just FYI: OsmAnd also shows fixme tags as a violet text on the map when the OSM module is enabled.

Reply:
> "Is its use now restricted to certain road users?"

For roads in cities in many cases the are limitations like maxweight, hgv ban. I worry that it may result in a technically correct answer leaving access=no foot=yes.

Overall, I expect adding fixme to be fairly rare, not happening for a typical highway=construction - so implementing (and translating) interface showing access data would be waste of time.

So I would prefer to avoid both simplified solution and "show all access data for confirmation".

I will check ratio of construction quests affected by fixme - if it is below say 2.5% I think that leaving fixme is fine.

Reply:
I'd very much prefer to either
1. exclude any highway constructions with an `access=X` tag set because of the uncertainty whether it refers to the current situation or the situation after the construction is complete 
2. ignore any access-tagging altogether, keep it as is, arguing that determining the access is not and should not be part of construction site -> is now finished process.


And document in the wiki that an access tag should not be set on a construction site because it is not clear whether it refers to the current situation or the situation when the construction is finished. Or, if set, then it should refer to the situation how it will be after the construction has finished because setting something like `access=no` on a construction site makes no sense whatsoever as it is implied and also there is nothing for data consumers to get confused about since it is tagged as `highway=construction + construction=X` (and not `highway=X + construction=yes` or something like that).

Reply:
Ok, I will remove handling of access.